mock/py/mock.py

```diff
@@ -381,17 +375,26 @@ def command_parse():
 
     # Bootstrap options
     parser.add_option('--bootstrap-chroot', dest='bootstrapchroot', action='store_true',
-                      help="build in two stages, using chroot rpm for creating the build chroot")
+                      help="build in two stages, using chroot rpm for creating the build chroot",
+                      default=None)
     parser.add_option('--no-bootstrap-chroot', dest='bootstrapchroot', action='store_false',
-                      help="build in a single stage, using system rpm for creating the build chroot")
+                      help="build in a single stage, using system rpm for creating the build chroot",
+                      default=None)
 
     parser.add_option('--use-bootstrap-image', dest='usebootstrapimage', action='store_true',
                       help="create bootstrap chroot from container image (turns "
-                           "--bootstrap-chroot on)")
+                           "--bootstrap-chroot on)", default=None)
     parser.add_option('--no-bootstrap-image', dest='usebootstrapimage', action='store_false',
-                      help="don't create bootstrap chroot from container image")
+                      help="don't create bootstrap chroot from container image", default=None)
+
+    (options, args) = parser.parse_known_args()
 
-    (options, args) = parser.parse_args()
+    # Optparse.parse_args() eats '--' argument, while argparse doesn't.  Do it manually.
+    if args and args[0] == '--':
+        args = args[1:]
```

Comment:
??? https://docs.python.org/3/library/argparse.html#arguments-containing

```
>>> import argparse
>>> parser = argparse.ArgumentParser(prog='PROG')
>>> parser.add_argument('baz', nargs='*')
_StoreAction(option_strings=[], dest='baz', nargs='*', const=None, default=None, type=None, choices=None, help=None, metavar=None)
>>> parser.parse_args(['--', '-x'])
Namespace(baz=['-x'])
```

Reply:
Better example is this:
```
In [14]: parser = argparse.ArgumentParser(prog='PROG')                                                              

In [15]: parser.add_argument('-x')                                                                                  
Out[15]: _StoreAction(option_strings=['-x'], dest='x', nargs=None, const=None, default=None, type=None, choices=None, help=None, metavar=None)

In [16]: parser.parse_known_args(['--'])                                                                            
Out[16]: (Namespace(x=None), ['--'])
```




```